frontend/lib/src/components/transaction_component/transaction_component.dart

```diff
@@ -94,120 +144,130 @@ class TransactionComponent extends StatelessWidget {
         } else if (transactionResponse.runtimeType == String) {
           alertDialog(content: transactionResponse);
         } else {
-          alertDialog();
+          alertDialog(content: transactionResponse.toString());
         }
 
         return;
       }
     }
 
-    return SingleChildScrollView(
-      padding: const EdgeInsets.fromLTRB(25, 30, 25, 0),
-      child: SizedBox(
-        height: 450,
-        child: GetX<TransactionController>(
-          builder: (_) => FormBuilder(
-            key: formKey,
-            autovalidateMode: AutovalidateMode.disabled,
-            onChanged: () {
-              formKey.currentState!.save();
-
-              // Fields required for each type
-              final Map<TransactionTypes, List<String>> requiredFields = {
-                TransactionTypes.CREDIT: ['amount', 'description'],
-              };
-
-              // Enable submit button once requirements are met
-              final validFields = requiredFields[type]
-                  ?.where(
-                    (field) => formKey.currentState!.fields[field]!.isValid,
-                  )
-                  .toList();
-
-              transactionController.setTransactionSubmitEnabled =
-                  requiredFields[type]?.length == validFields?.length;
-            },
-            child: Column(
-              children: [
-                Text(
-                  label,
-                  style: Theme.of(context).textTheme.labelLarge!.copyWith(
-                        fontWeight: FontWeight.bold,
-                      ),
-                ),
-                const SizedBox(
-                  height: 13,
-                ),
-                if (type == TransactionTypes.TRANSFER) ...[
-                  InputField(
-                    name: 'email',
-                    label: 'Email',
-                    inputType: TextInputType.emailAddress,
-                    validator: FormBuilderValidators.email(),
+    return GetX<TransactionController>(
```

Comment:
refactor, put GetX as close to as the using of the controller

currently its
Getx -> scroll -> sizedbox -> formbuilder -> reference

wrap formbuilder instead